src/Mcamara/LaravelLocalization/LaravelLocalization.php

```diff
@@ -16,61 +16,27 @@
 
 class LaravelLocalization
 {
-    /**
-     * Illuminate request class.
-     *
-     * @var string
-     */
-    protected $baseUrl;
-
-    /**
-     * Default locale.
-     *
-     * @var string
-     */
-    protected $defaultLocale;
-
-    /**
-     * Supported Locales.
-     *
-     * @var array
-     */
-    protected $supportedLocales;
-
-    /**
-     * Locales mapping.
-     *
-     * @var array
-     */
-    protected $localesMapping;
-
-    /**
-     * Current locale.
-     *
-     * @var string|false
-     */
-    protected $currentLocale = false;
+    protected string $baseUrl;
+    protected string $defaultLocale;
+    protected array $supportedLocales;
+    protected array $localesMapping;
+    protected string | false $currentLocale = false;
```

Comment:
I assume `false` is intended to indicate that `$currentLocale` has not been set. In this case, `null` would likely be a better fit. However, `$currentLocale` is used in multiple functions, and most of them assume it's always a string without performing type checks. As a result, they fail if `$currentLocale` is unset, whether its value is `string|false` or `string|null` or just `string`.

Given the current implementation, I would suggest sticking with:

```php
protected string $currentLocale;
```

That said, the overall handling of `$currentLocale` is problematic. The need to declare and manage it separately before calling its related functions makes the codebase unnecessarily fragile. Since `$currentLocale` always mirrors the value of `\App::getLocale()`, it might be best to eliminate it altogether and rely directly on `\App::getLocale()`.

Reply:
@iwasherefirst2 I agree with your points here. There's a couple of places where `false` is used when `null` would have been a better choice. But I'd tackle this in a different PR, as this one just focuses on adding types.

Reply:
Sure